### ISSUE TITLE
Broken rule: interface-name-prefix

### ISSUE THREAD

**Issue by corbinu** [nzakas/eslint-plugin-typescript#29]
Looks like some of the changes in either the parser or eslint 4 broke one of the rules here.

It appears to be when trying to parse any interface

@soda0289 Thoughts?

- eslint: 4.0.0-rc1
- plugin version: @corbinu/eslint-plugin-typescript@0.2.0
- parser: 3.0.0

```
Cannot read property 'name' of undefined
TypeError: Cannot read property 'name' of undefined
    at Linter.checkInterfacePrefix (/home/corbinu/Developer/Decimal/modules/google-distance/node_modules/@corbinu/eslint-plugin-typescript/lib/rules/interface-name-prefix.js:62:55)
    at emitOne (events.js:115:13)
    at Linter.emit (events.js:210:7)
    at NodeEventGenerator.applySelector (/home/corbinu/Developer/Decimal/modules/google-distance/node_modules/eslint/lib/util/node-event-generator.js:265:26)
    at NodeEventGenerator.applySelectors (/home/corbinu/Developer/Decimal/modules/google-distance/node_modules/eslint/lib/util/node-event-generator.js:294:22)
    at NodeEventGenerator.enterNode (/home/corbinu/Developer/Decimal/modules/google-distance/node_modules/eslint/lib/util/node-event-generator.js:308:14)
    at CodePathAnalyzer.enterNode (/home/corbinu/Developer/Decimal/modules/google-distance/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:602:23)
    at Traverser.enter (/home/corbinu/Developer/Decimal/modules/google-distance/node_modules/eslint/lib/linter.js:920:36)
    at Traverser.__execute (/home/corbinu/Developer/Decimal/modules/google-distance/node_modules/estraverse/estraverse.js:397:31)
    at Traverser.traverse (/home/corbinu/Developer/Decimal/modules/google-distance/node_modules/estraverse/estraverse.js:501:28)
```

**Comment by soda0289** [comment]
There have been several breaking changes recently to typescript-eslint-parser. I'm not surprised the plugin has stopped working. The typescript interface node has changed completly that is probably what is causing this error. I will take a look this weekend and try to fix everything.

**Comment by corbinu** [comment]
@soda0289 Thanks take your time just wanted to let you know. Also did you want me to add you with publishing writes on my npm package for this repo?

**Comment by soda0289** [comment]
Yea sure, I use the same name as github. I'm still not sure if we should fork this whole repo or just publish the package under your prefix, but I think for now we should continue to use this repo.

**Comment by corbinu** [comment]
Yes for sure if it we need I am totally ok with just renaming my fork something else but really hoping we don't have to.

**Comment by corbinu** [comment]
@soda0289 What if I publish my fork as eslint-plugin-ts. We can still use this repo for now and switch over to mine if we have to. Worst case we mark eslint-plugin-ts as deprecated and redirect back to eslint-plugin-typescript

**Comment by soda0289** [comment]
Yea that might work better, we could also use an organization name like `eslint-typescript`, but I think its simpler to just use `eslint-plugin-ts`. I'll post some PR's soon and bump the version as well in this repo.

Thanks for all the help!

**Comment by corbinu** [comment]
No problem just wish I had more time to help more

**Comment by JamesHenry** [comment]
@corbinu was this addressed by your PR? (and therefore fixed in 0.3.0)

**Comment by corbinu** [comment]
@JamesHenry Yep all existing rules now work :)
